tests/v2/actions/ssh/ssh.go

```diff
@@ -0,0 +1,126 @@
+package ssh
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/v2/actions/clusters"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/kubectl"
+	"github.com/rancher/shepherd/extensions/sshkeys"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/nodes"
+	"github.com/sirupsen/logrus"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extensionsClusters "github.com/rancher/shepherd/extensions/clusters"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	NodeListEmptyMessageError = "node list is empty"
+	ExternalIPAnnotation      = "rke.cattle.io/external-ip"
+)
+
+// CreateSSH is a helper to create a SSH Node
+func CreateSSH(client *rancher.Client, steveClient *steveV1.Client, clusterName string, clusterID string) (*nodes.Node, error) {
+
+	provisioningClusterID, err := extensionsClusters.GetV1ProvisioningClusterByName(client, client.RancherConfig.ClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster, err := client.Steve.SteveType(extensionsClusters.ProvisioningSteveResourceType).ByID(provisioningClusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	newCluster := &provv1.Cluster{}
+	err = steveV1.ConvertToK8sType(cluster, newCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	sshNode := &nodes.Node{}
+	if strings.Contains(newCluster.Spec.KubernetesVersion, "rke2") || strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") {
```

Comment:
[nit] usually we should avoid having parent function handling logic for rke2/k3s + rke1, but in this function I am fine making an exception. I would ask that you split the contents of the if statement (and separately the else statement) into private helpers that each return an (sshNode, err) to break down the read complexity of this function a bit